viewer/src/cache/MemoryRequestCache.ts

```diff
@@ -27,59 +27,58 @@ class TimestampedContainer<T> {
   }
 }
 
-export type RemoveDelegate<Key, Result> = (key: Key, result: Result) => void;
-
-export interface MemoryRequestCacheOptions<Key, Result> {
+export interface MemoryRequestCacheOptions {
   maxElementsInCache?: number;
-  remove?: RemoveDelegate<Key, Result>;
+  // remove?: RemoveDelegate<Key, Result>;
 }
 
-export class MemoryRequestCache<Key, Data, Result> implements RequestCache<Key, Data, Result> {
+export class MemoryRequestCache<Key, Data> implements RequestCache<Key, Data> {
   private readonly _maxElementsInCache: number;
-  private readonly _results: Map<Key, TimestampedContainer<Result>>;
-  private readonly _request: RequestDelegate<Key, Data, Result>;
-  private readonly _removeCallback?: RemoveDelegate<Key, Result>;
+  private readonly _data: Map<Key, TimestampedContainer<Data>>;
 
-  constructor(request: RequestDelegate<Key, Data, Result>, options?: MemoryRequestCacheOptions<Key, Result>) {
-    this._results = new Map();
-    this._request = request;
+  constructor(options?: MemoryRequestCacheOptions) {
+    this._data = new Map();
     this._maxElementsInCache = (options && options.maxElementsInCache) || 50;
-    this._removeCallback = options && options.remove;
   }
 
-  request(id: Key, data: Data): Result {
-    const existing = this._results.get(id);
-    if (existing) {
-      return existing.value;
-    }
-
-    this.cleanupCache();
-    const result = new TimestampedContainer<Result>(this._request(id, data));
-    this._results.set(id, result);
-    return result.value;
+  has(id: Key) {
+    return this._data.has(id);
   }
 
-  clearCache() {
-    this._results.clear();
+  add(id: Key, data: Data) {
+    if (this._data.size < this._maxElementsInCache) {
+      this._data.set(id, new TimestampedContainer(data));
+    } else {
+      throw new Error('Cache full, please clean Cache and retry adding data');
+    }
   }
 
-  private cleanupCache() {
-    const maxCacheSize = this._maxElementsInCache;
-    if (this._results.size > maxCacheSize) {
-      const allResults = Array.from(this._results.entries());
-      allResults.sort((left, right) => {
-        return right[1].lastAccessTime - left[1].lastAccessTime;
-      });
+  get(id: Key): Data {
+    const data = this._data.get(id);
+    if (data !== undefined) {
+      // Don't really like the touch for lastTime being hidden within a get function. Should we maybe make a
+      // TimeConstrainedCache interface where the geter is called something like getAndUpdateTimestamp for clarity?
```

Comment:
Why does the outside world need to worry about it? It's an implementation detail of MemoryRequestCache, but not part of the RequestCache-contract